src/Registration/pReg/Reg.py

```diff
@@ -520,6 +520,13 @@ def are_equal_to_given_accuracy(im1, im2, accuracy):
         pyiutil.deleteDataHandle(h)
         return value
 
+    @property
+    def shape(self):
+        return self.dimensions()
+    @property
+    def dtype(self):
+        return numpy.float32
+
```

Comment:
Would it not be enough to have this up in `DataContainer`, `dtype` to be overrided by MR `dtype`?

Same for all PET containers of course.